packages/runtime/src/runtime.ts

```diff
@@ -1161,4 +1166,50 @@ export class Runtime {
 	sendTxAndWait(transactions: SignedTransaction[]): TxnReceipt[] {
 		return this.executeTx(transactions);
 	}
+
+	/**
+	 * Produces new block and adds it to a Map where the keys are block numbers
+	 */
+	produceBlock() {
+		let timestamp: bigint | undefined;
+		let seed: Uint8Array | undefined;
+		if (this.store.blocks.size === 0) { //create genesis block
+			seed = nacl.randomBytes(32);
+			timestamp = BigInt(Math.round(new Date().getTime() / 1000));
+		} else { //add another block
+			const lastBlock = this.store.blocks.get(this.round);
+			if (lastBlock !== undefined) {
+				seed = new TextEncoder().encode(MD5(lastBlock.seed.toString()).toString());
+				timestamp = lastBlock.timestamp + BlockFinalisationTime;
+				this.round += 1;// we move to new a new round
+			}
+		}
+		if (timestamp !== undefined && seed !== undefined) {
+			this.store.blocks.set(this.round, { timestamp, seed });
+		} else {
+		throw new RuntimeError(RUNTIME_ERRORS.GENERAL.PRODUCE_BLOCK); 
+		}
+	}
+	/**
+	 * Returns a requested Block object. If it does not exist on chain throws an error
+	 * @param round block number
+	 * @returns Block
+	 */
+	getBlock(round: number): Block {
+		const block = cloneDeep(this.store.blocks.get(round));
+		if (block !== undefined) {
```

Comment:
```suggestion
		if (block) {
```
undefined is always false. try to avoid these type of checks.